### ISSUE TITLE
Increase minification: emit_attributes() should pick single or double quotation form for escaping

### ISSUE THREAD

**Issue by raphaelcohn** [netvl/xml-rs#164]
A nice feature to improve minification would to let `emit_attributes()` choose to use either a single or double quoted form for attribute values depending on the presence of double or single quotes in the attribute value to be escaped. This obviously has a performance hit, so it should be an option in `EmitterConfig`.

**Comment by netvl** [comment]
Good idea! Pull requests are of course welcome. Not sure if I will be able to work on this any time soon.
